### PR TITLE
Issue 18249 refactor dot field variables

### PR DESCRIPTION
**Pull request by alfredo-dotcms**

### CODE REVIEW

File: src/app/portlets/shared/dot-content-types-edit/components/fields/dot-content-type-fields-variables/dot-content-type-fields-variables.component.ts [hunk 1]
```diff
@@ -70,75 +37,23 @@ export class DotContentTypeFieldsVariablesComponent implements OnInit, OnChanges
...
                 () => {
-                    [this.fieldVariables, this.fieldVariablesBackup] = [
-                        this.fieldVariables,
-                        this.fieldVariablesBackup
-                    ].map((variables: DotFieldVariable[]) => {
-                        return variables.filter(
-                            (_item: DotFieldVariable, index: number) => index !== fieldIndex
-                        );
-                    });
+                    [this.fieldVariables] = [this.fieldVariables].map(
```
Comment (fmontes) [inline src/app/portlets/shared/dot-content-types-edit/components/fields/dot-content-type-fields-variables/dot-content-type-fields-variables.component.ts:50]: Shouldn't this be a filter?

File: src/app/view/components/dot-key-value/dot-key-value.component.html [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    #table
+    [value]="variablesBackup"
+    sortField="clazz"
+    (keydown.enter)="$event.stopPropagation()"
+    sortOrder="0"
+    dataKey="id"
+>
+    <ng-template pTemplate="header" let-columns>
+        <tr>
+            <th [ngStyle]="{ 'text-align': 'left', width: '33%' }">
```
Comment (fmontes) [inline src/app/view/components/dot-key-value/dot-key-value.component.html:11]: Can we move this styles to a file and use variables?

File: src/app/view/components/dot-key-value/dot-key-value.component.html [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+            [variablesList]="variables"
+            (cancel)="onCancel($event)"
+            (delete)="deleteVariable($event)"
+            (save)="saveVariable($event)"
+        >
+        </dot-key-value-table-row>
+    </ng-template>
+    <ng-template pTemplate="emptymessage" let-variables>
+        <tr>
+            <td [attr.colspan]="3">
```
Comment (fmontes) [inline src/app/view/components/dot-key-value/dot-key-value.component.html:36]: `[attr.colspan]` is when you have to do some sort of dynamic value but templates takes regular HTML so...

```suggestion
            <td colspan="3">
```

**Comment by alfredo-dotcms** [comment]
this refactor needs changes to be added soon on a new PR

**Pull request closed without merging**